### PR TITLE
Link and adapt to plugwise v1.6.0

### PR DESCRIPTION
**Pull request by bouwew**
## Release Notes
- **BREAKING change for Adam**: the existing device-based climate entities will be replaced by new zone-based climate entities most likely with a different name.

- **New Features**
  - Introduce zone-based climate entities for the Adam device, replacing device-based climate entities.
  - Add available zone-based sensors.
  - Reboot button for compatible gateways added to user controls.

- **Bug Fixes**
  - Adam: climate entities are now based on zones. Being based on a single device was incorrect.

- **Documentation**
  - Updated README for clarity on device support and usage, including emphasis on reading release notes before upgrades.
  - Enhanced changelog with new version details and feature descriptions.

- **Tests**
  - Updated test cases to align with new entity naming conventions and structures.

### CODE REVIEW

File: custom_components/plugwise/binary_sensor.py [hunk 1]
```diff
@@ -174,12 +174,12 @@ def is_on(self) -> bool:
...
                 )
 
-        return self.device[BINARY_SENSORS][self.entity_description.key]
+        return self.pw_entity[BINARY_SENSORS][self.entity_description.key]
 
     @property
     def extra_state_attributes(self) -> Mapping[str, Any] | None:
         """Return entity specific state attributes."""
-        if self.entity_description.key != PLUGWISE_NOTIFICATION:  # Upstream const
+        if self.entity_dscription.key != PLUGWISE_NOTIFICATION:  # Upstream const
```
[the remark this answers is not in the document]
Comment (bouwew) [reply custom_components/plugwise/binary_sensor.py]: Fixed

File: custom_components/plugwise/coordinator.py [hunk 2]
```diff
@@ -137,18 +137,18 @@
...
 
         # Then remove the connected orphaned device(s)
-        for device_entry in device_list:
+        for device_entry in pw_entities_list:
             for identifier in device_entry.identifiers:
                 if identifier[0] == DOMAIN:
                     if (
                         device_entry.via_device_id == via_device_id
-                        and identifier[1] not in data.devices
+                        and identifier[1] not in data.device_zones
```
[the remark this answers is not in the document]
Comment (bouwew) [reply custom_components/plugwise/coordinator.py]: Fixed

File: custom_components/plugwise/coordinator.py [hunk 3]
```diff
@@ -110,24 +110,24 @@ async def _async_update_data(self) -> PlugwiseData:
             raise ConfigEntryError("Device with unsupported firmware") from err
         else:
             LOGGER.debug(f"{self.api.smile_name} data: %s", data)
-            await self.async_add_remove_devices(data, self.config_entry)
+            await self.async_add_remove_plugise_entities(data, self.config_entry)
```
[the remark this answers is not in the document]
Comment (bouwew) [reply custom_components/plugwise/coordinator.py]: Fixed

File: custom_components/plugwise/binary_sensor.py [hunk 4]
```diff
@@ -130,16 +130,16 @@ def _add_entities() -> None:
...
-            if not (binary_sensors := device.get(BINARY_SENSORS)):
+        for pw_entity_id in coordinator.new_pw_entities:
+            pw_entity = coordinator.data.entities[pw_entity_id]
+            if not (binary_sensors := pw_entity.get(BINARY_SENSORS)):
                 continue
             for description in PLUGWISE_BINARY_SENSORS:
                 if description.key not in binary_sensors:
                     continue
-                entities.append(PlugwiseBinarySensorEntity(coordinator, device_id, description))
+                ha_entities.append(PlugwiseBinarySensorEntity(coordinator, pw_entity, description))
```
[the remark this answers is not in the document]
Comment (bouwew) [reply custom_components/plugwise/binary_sensor.py]: Fixed

File: custom_components/plugwise/climate.py [hunk 5]
```diff
@@ -303,11 +322,11 @@ async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
                 await self.async_set_preset_mode(PRESET_AWAY)  # pragma: no cover
             if (
                 self._homekit_mode in [HVACMode.HEAT, HVACMode.HEAT_COOL]
-                and self.device[ACTIVE_PRESET] == PRESET_AWAY
+                and self.device_zone[ACTIVE_PRESET] == PRESET_AWAY
             ):  # pragma: no cover
                 await self.async_set_preset_mode(PRESET_HOME)  # pragma: no cover
```
[the remark this answers is not in the document]
Comment (bouwew) [reply custom_components/plugwise/climate.py]: Fixed

File: custom_components/plugwise/select.py [hunk 6]
```diff
@@ -83,28 +83,21 @@ async def async_setup_entry(
...
-        #     PlugwiseSelectEntity(coordinator, device_id, description)
-        #     for device_id in coordinator.new_devices
-        #     for description in SELECT_TYPES
-        #     if description.options_key in coordinator.data.devices[device_id]
-        # )
-        # pw-beta alternative for debugging
-        entities: list[PlugwiseSelectEntity] = []
-        for device_id in coordinator.new_devices:
-            device = coordinator.data.devices[device_id]
+        entities: list[PlugwiseClimateEntity] = []
```
[the remark this answers is not in the document]
Comment (bouwew) [reply custom_components/plugwise/select.py]: Implemented.

File: custom_components/plugwise/binary_sensor.py [hunk 7]
```diff
@@ -174,12 +174,12 @@ def is_on(self) -> bool:
                     self.hass, message, "Plugwise Notification:", f"{DOMAIN}.{notify_id}"
                 )
 
-        return self.device[BINARY_SENSORS][self.entity_description.key]
+        return self.pw_entityu[BINARY_SENSORS][self.entity_description.key]
```
[the remark this answers is not in the document]
Comment (bouwew) [reply custom_components/plugwise/binary_sensor.py]: Fixed

**Review by CoMPaTech — APPROVED** [review]

**Pull request merged**